src/libraries/ArrayUtil.sol

```diff
@@ -120,9 +132,9 @@ library ArrayUtil {
         if (z >= x.length) return x;
         y = new uint256[](x.length - 1);
         for (uint256 i; i < x.length; ) {
+            unchecked {
```

Comment:
Lets not wrap everything in `unchecked`

Reply:
The if/else needs to be in the same unchecked block in order to stop i-1 from being checked. y[i-1] should never underflow